boards: cyw920829m2evk: flash configuration moved to board

MerinMGeorge opened the pull request:
- CYW920829 configuration from the flash driver is moved to modules hal_infineon zephyr-ifx-cycfg folder

MerinMGeorge committed:
drivers: flash: cyw920829 configuration moved from driver to boards

 - Flash configuration was previously done in the driver selected by
   Kconfig. Instead move flash configuration to the board removing
   the need for ifdefs in the driver. This will allow boards to
   configure specifics related to the on board (not part of the SoC)
   flash.

Signed-off-by: Merin George <merin.george@infineon.com>

merin-infineon commented on modules/hal_infineon/CMakeLists.txt:
```diff
@@ -113,3 +113,7 @@ if(CONFIG_SOC_FAMILY_INFINEON_PSOC4)
 endif()
+
+if(CONFIG_SOC_SERIES_CYW20829)
+  add_subdirectory(zephyr-ifx-cycfg)
```
Fixed

teburd commented:
To be clear, the changes are good, needs to pass CI still to get approved and merged.

teburd reviewed (commented):
Commit message needs a little work. The title should match the addition not the subtraction. The message should cover the what and the why.

For example something like...

```
boards: cyw920829m2evk: Moves flash configuration from driver to board

Flash configuration was previously being done in the driver selected by Kconfig. Moves flash configuration to the board removing the need for ifdefs in the driver. This will allow boards to configure specifics related to the on board (not part of the SoC) flash.

Signed-off-by: Merin George <merin.george@infineon.com>
```

We ideally want to see in the commit message what you are changing and why.

MerinMGeorge committed:
Kconfig: fix the define to disable SysTick when LP timer is enabled

Fix the Kconfig dependency for CYW20829 so that enabling the
low-power system timer automatically disables the Cortext-M SysTick

Signed-off-by: Merin George <merin.george@infineon.com>
